apps/sms/views/inbox/js/inbox.js

```diff
@@ -405,7 +405,7 @@ var InboxView = {
       var thread = Threads.get(+id);
 
       var markable = thread && !thread.isDraft &&
-        (isRead || !thread.getDraft());
+        (isRead || !thread.getDraft()) && !(thread.unreadCount ^ isRead);
```

Comment:
@steveck-chung  I think , you meant
<code>
var readStatusMatched = (!!thread.unreadCount ^ isRead);
</code>